client/src/main/scala/org/http4s/client/oauth1/Header.scala

```diff
@@ -0,0 +1,77 @@
+package org.http4s.client.oauth1
+
+import cats.Show
+import cats.kernel.Order
+import cats.implicits._
+sealed trait Header {
```

Comment:
Is there any reason not to model these as standard `org.http4s.Header`?  We can still have the specific types, but these won't work quite the same as other headers in http4s.

Reply:
I see it now in `takeSigHeaders`.  Yeah, maybe make this a marker trait, but still make these standard http4s headers?

Reply:
I'm not sure these are standard headers, these are part of. So i'm not sure if we want them to be Http header.

```Authorization: OAuth realm="Example",
  oauth_consumer_key="0685bd9184jfhq22",
  oauth_token="ad180jjd733klru7",
  oauth_signature_method="HMAC-SHA1",
  oauth_signature="wOJIO9A2W5mFwDgiDvZbTSMK%2FPY%3D",
  oauth_timestamp="137131200",
  oauth_nonce="4572616e48616d6d65724c61686176",
  oauth_version="1.0"
```

Reply:
You're right.  I got fooled by the name.  These are called "parameters" in most headers, and the spec refers to them as "protocol parameters".  Maybe I'd call the trait `ProtocolParameter` to avoid ambiguity with `Header`.